src/api/app/models/token/workflow.rb

```diff
@@ -0,0 +1,71 @@
+class Token::Workflow < Token
+  def self.token_name
+    'workflow'
+  end
+
+  def call(options)
+    payload = options[:payload]
+    scm = options[:scm] # github, gitlab
+    event = options[:event] # pull_request, merge_request
+
+    extractor = TriggerControllerService::ScmExtractor.new(scm, event, payload)
+
+    return unless extractor.accepted_event_and_action?
+
+    ['Event::BuildFail', 'Event::BuildSuccess'].each do |build_event|
+      EventSubscription.create!(eventtype: build_event,
+                                receiver_role: 'watcher', # TODO: check if this makes sense
+                                user: user,
+                                channel: 'scm',
+                                enabled: true,
+                                token: self,
+                                payload: payload)
+    end
+
+    SCMStatusReporter.new(extractor.extract, scm_token).call
+    # scm_extractor_payload = extractor.extract # returns { scm: 'github', repo_url: 'http://...' }
+
+    # yaml_file = Workflows::YAMLDownloadService.new(scm_extractor_payload).call
+
+    # Read configuration file
+    #   if the ref is not included in the config file's branch whitelist we do nothing.
+
+    # Decide what to do by action:
+    # FIXME: Who does this branching? Should we just overwrite the _service file with the data coming
+    #        from the workflow config file and trigger a Service token?
```

Comment:
Where is the documentation for that `_branch_request` file?

Reply:
@adrianschroeter yeah we are going to use this. We should talk about the format though, I think it would be more consistent if this would be (more specific) XML and not a json dump.